x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -776,7 +776,20 @@ public static RoleDescriptor kibanaSystemRoleDescriptor(String name) {
                         ".metrics-endpoint.metadata_united_default"
                     )
                     .privileges("create_index", "delete_index", "read", "index")
-                    .build(), },
+                    .build(),
+                    // For src/dest indices of the CSP packages that ships a transform
+                    RoleDescriptor.IndicesPrivileges.builder()
+                        .indices("logs-cis_kubernetes_benchmark.findings-*")
+                        .privileges("read", "view_index_metadata")
+                        .build(),
+                    RoleDescriptor.IndicesPrivileges.builder()
+                        .indices(
+                            "cloud_security_posture-findings_latest",
+                            "cloud_security_posture-benchmark_scores"
+                        )
+                        .privileges("create_index", "delete_index", "read", "index", "delete" ,"write" ,"all")
```

Comment:
Do we need `delete`? I don't see it as a requirement for the transform APIs nor do I expect Kibana to be deleting documents from these indices

Reply:
@elastic/ml-data (I hope this is the right team) are our docs out of date here? I'm surprised that installing a transform is attempting a delete by query on the destination index and that the docs don't indicate this. I'm also surprised this issue doesn't affect our other transforms installed by Fleet?

> ```
> task encountered irrecoverable failure: org.elasticsearch.ElasticsearchSecurityException: action [indices:data/write/delete/byquery] is unauthorized for user [kibana_system] with roles [kibana_system] on indices [cloud_security_posture-findings_latest], this action is granted by the index privileges [delete,write,all]
> ```

That error seems to indicate that they need one of these `[delete,write,all]` privileges, but not all of them. Can we test what the minimum required here is?

Reply:
Hi @joshdover - I think @elastic/ml-ui would be more help here

Reply:
@joshdover we need delete by query. This is for the new data retention policies that transforms can create. Our docs our out of date if delete is not included.

Reply:
I will create an ES issue to capture this inaccuracy and adding a check early on to verify security (like we do for write/read).

Reply:
https://github.com/elastic/elasticsearch/issues/85409 for tracking our missing validation and documentation.

Reply:
Thank you all, I added the `delete` privilege.